### DISCUSSION TITLE
Creating a scatterplot using Deepscatter

### DISCUSSION THREAD

**Discussion by chaz23** [observablehq/framework#1881] [Q&A]
I'm a bit stuck trying to create a scatterchart using Deepscatter. I've tried to work out what's going on [here](https://observablehq.com/@davidnmora/intro-to-deepscatter) and recreate it in Framework but have been failing to load my own data. I'd really appreciate it if someone could show me a minimal working example to get me started! This is what I adapted from the example:

**index.md:**

```js
import Deepscatter from "npm:deepscatter";
```

<div id="plot" style="min-width:300px;min-height:600px"></div>

```js
new Deepscatter("#plot", 300, 600)
  .plotAPI({
    // Data fetched from Ben's website. You could also specify a local file, too.
    source_url: "https://benschmidt.org/arxiv",
    background_color: "#ffffff",
    max_points: 10000,
    point_size: 16,
    alpha: 5,
    encoding: {
      x: {
        field: "x",
      },
      y: {
        field: "y",
      },
      color: {
        constant: "#00ff00",
      },
    },
  })
  .then(() => {
    console.log("... chart update done!");
  });
```

This works, however I want to use my own data, and can't manage to get it right...

**R data loader (./data/test.parquet.R):**

```r
library(dplyr)
library(arrow)

output <- starwars %>% select(mass, height)

temp_file <- tempfile(fileext = ".parquet")
write_parquet(output, sink = temp_file)

system2('cat', args = temp_file)
```

How do I modify the `source_url` parameter in `plotAPI` to point to my data?

**Comment by Fil** [comment]
I assume it would be

```
source_url:  FileAttachment("./data/test.parquet").href
```

**Reply by chaz23** [reply]
That didn't seem to work...
![image](https://github.com/user-attachments/assets/662099bf-e2d7-417c-887a-08fe11f78236)

**Comment by Fil** [comment]
It appears that Deepscatter expects a "zoomable" directory of files in the feather format, not a parquet file.
